packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -327,30 +314,35 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
             commits[decodeURIComponent(key)] = hierarchicalTree.commits[key];
         }
 
+        let finalTree: api.ISnapshotTree | undefined;
         if (commits && commits[".app"]) {
             // The latest snapshot is a summary
             // attempt to read .protocol from commits for backwards compat
-            return this.readSummaryTree(tree.id, commits[".protocol"] || hierarchicalTree.trees[".protocol"], commits[".app"] as string);
+            finalTree = await this.readSummaryTree(tree.id, commits[".protocol"] || hierarchicalTree.trees[".protocol"], commits[".app"] as string);
         }
 
-        if (hierarchicalTree.blobs) {
-            const attributesBlob = hierarchicalTree.blobs.attributes;
-            if (attributesBlob) {
-                this.attributesBlobHandles.add(attributesBlob);
+        if (finalTree === undefined) {
```

Comment:
Was this supposed to be unconditional else statement?
I.e. before, the only thing we did in readSummaryTree was buildCachesForDedup and then return.
All the code below was for the case where we did not take if (commits && commits[".app"]) branch.

Other way to ask it - isn't readSummaryTree() returning undefined a failure that needs to result in this function returning a failure?

Unrelated though: it's a bit hard sometimes to follow code as it contains cases for very old behaviors. For example, the check for if (commits && commits[".app"]) suggests that this should be always true for summaries, and not true for snapshot that we have not written for 1.5 years. If that's true, should be delete support for snapshots? There could be a file that was never opened in last 1.5 years, and in future we would need to support those forever, but given we are talking about pre-production (and even pre- pre-release), I think it's Ok not to stop supporting such paths in favor of making code more simpler / straightforward. This is only true if my analyzes (about code and dates) is correct.

Reply:
Yes, good catch. Added it in else statement. For other questions, it is not for backward compat. Also added comment in code.

        // For container loaded from detach new summary, we will not have a commit for ".app" in downloaded summary as the client uploaded both ".app" and ".protocol" trees by itself. For other summaries, we will have ".app" as commit because client previously only uploaded the app summary.